sherpa/astro/ui/tests/test_astro_session.py

```diff
@@ -4317,3 +4320,249 @@ def test_dataspace1d_datapha_invalid_args(start, stop, step, numbins):
     with pytest.raises(DataErr):
         s.dataspace1d(start, stop, step=step, numbins=numbins,
                       dstype=DataPHA)
+
+
+# Ideally there would be a test of the low-level code, but it's
+# important to check all the pieces tie together in the UI layer.
+#
+def setup_multicore_data(s: Session) -> tuple[ArithmeticModel, ArithmeticModel]:
+    """Create data and model for fitting.
+
+    The idea is to have a complex-enough dataset for fitting.  It
+    should really be fit with a Poisson statistic, but the meaning of
+    the data is secondary here to the mechanics of the process.
+
+    """
+
+    # Use a known random generator that is unlikely to change.
+    #
+    rng = np.random.RandomState(28567883)
+    x = np.linspace(10, 20, 101)
+
+    cpt1 = s.create_model_component("pseudovoigt1d", "cpt1")
+    cpt2 = s.create_model_component("pseudovoigt1d", "cpt2")
+
+    cpt1.pos = 14
+    cpt2.pos = 16
+    cpt1.fwhm = 2
+    cpt2.fwhm = 1
+    cpt2.ampl = 0.5
+
+    y = 40 * (cpt1(x) + cpt2(x))
+    ynoise = poisson_noise(y, rng=rng)
+
+    s.load_arrays(1, x, ynoise)
+
+    # Set up the model
+    g1 = s.create_model_component("gauss1d", "g1")
+    g2 = s.create_model_component("gauss1d", "g2")
+    s.set_model(g1 + g2)
+
+    g1.pos = 13
+    g2.pos = 17
+    g1.fwhm = 1
+    g2.fwhm = 1
+
+    # Scale the normalization to match the data. Note that sherpa.ui does
+    # not have calc_data_sum / calc_model_sum.
+    #
+    dsum = ynoise.sum()
+    msum = (g1 + g2)(x).sum()
+    g1.ampl.val = dsum / msum
+    g2.ampl.val = dsum / msum
+
+    return g1, g2
+
+
+# Note: gridsearch is too slow to test this way
+#
+@pytest.mark.skipif(not multi, reason="multi-core support not enabled")
+@pytest.mark.parametrize("session", [Session, AstroSession])
+@pytest.mark.parametrize("ncores", [1, 2, 3])
+def test_method_numcores_levmar(session, ncores):
```

Comment:
This test is also redundant with `test_errors_numcores_levmar_proj` which also generates data and fits it (and then runs `proj` after that. In the interest of run time I suggest to merge the tests - just copy the assert statements for here into the other test and this one won't be needed any more.
One could argue similarly for the `conf` test, but maybe run one of them on `Session` and the other on `Astrosession` so that we have two separate tests for levmar in the end.

Due to the duplication and heavy parameterization, these new tests increase the CI runtime by almost 10%, which seems like a lot give how similar e.g. the code in `Session` and `AstroSession` is.